### PR TITLE
Brms 190 caseclass

### PR DESCRIPTION
**Pull request by manishsw**
@adamconder  this include changes of 167.

### CODE REVIEW

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala [hunk 1]
```diff
@@ -67,13 +68,16 @@ class BirthEventsControllerSpec
...
-        contentType(result.get).get shouldBe "application/json"
-        header(ACCEPT, result.get).get shouldBe "application/vnd.hmrc.1.0+json"
+        val request = postRequest(userMatchExcludingReferenceNumberKeyForScotland)
+        val result = MockController.post().apply(request)
 
-        jsonBodyOf(result.get).map(x => x should have('matched (false)))
+        status(result) shouldBe OK
+        contentType(result).get shouldBe "application/json"
+        header(ACCEPT, result).get shouldBe "application/vnd.hmrc.1.0+json"
+        //TODO need to change once response mapping done.
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: @manishsw is this comment required now?

**Review by adamconder — CHANGES_REQUESTED** [review]

**Comment by adamconder** [comment]
@manishsw happy overall! Good work. Nice implementation of the JSON Reads.
Need to resolve the merge conflicts before can be merged. Please pull from master.

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala (hunk 1 above)
Comment (manishsw) [reply test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: @adamconder  No, not required . removed it.

**Comment by manishsw** [comment]
@adamconder  done changes, kindly merge it.

**Review by adamconder — APPROVED** [review]

**Pull request merged**